File: src/etch/bindings/csharp/util/StrIntHashMap.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Util
{
    /// <summary>
    /// A hashmap which is from string to integer
    /// </summary>
    public class StrIntHashMap : Dictionary<string, int?>
    {
        public StrIntHashMap()
        {
            // nothing to do .
        }

        /// <summary>
        /// Constructs a StrIntHashMap initialized from another
        /// </summary>
        /// <param name="other"></param>
        public StrIntHashMap( StrIntHashMap other )
            : base( other )
        { }

    }
}

File: src/etch/bindings/java/util/StrIntHashMap.java
/* $Id$
 *
 * Created by wert on Jun 25, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.util;

import java.util.HashMap;

/**
 * A HashMap which is from String to Integer.
 */
public class StrIntHashMap extends HashMap<String, Integer>
{
	private static final long serialVersionUID = -9162231227512652630L;

	/**
	 * Constructs a StrIntHashMap initialized from another.
	 * @param other
	 */
	public StrIntHashMap( StrIntHashMap other )
	{
		super( other );
	}
	
	/**
	 * Constructs a StrIntHashMap.
	 */
	public StrIntHashMap()
	{
		// nothing to do.
	}
}
